astropy/time/core.py

```diff
@@ -282,6 +283,8 @@ def _get_time_fmt(self, val, val2, format, scale):
             try:
                 return FormatClass(val, val2, scale, self.precision,
                                    self.in_subfmt, self.out_subfmt)
+            except UnitConversionError:
```

Comment:
I think it is fine just to remove the additional exception. I don't see how `UnitConversionError` could indicate anything else than "this format doesn't work, go on try the next".